backend/dashboard/static/js/table_webcomponent.js

```
@@ -0,0 +1,43 @@
+(()=>{var b=Object.defineProperty;var p=Object.getOwnPropertySymbols;var w=Object.prototype.hasOwnProperty,g=Object.prototype.propertyIsEnumerable;var m=(o,t,e)=>t in o?b(o,t,{enumerable:!0,configurable:!0,writable:!0,value:e}):o[t]=e,h=(o,t)=>{for(var e in t||(t={}))w.call(t,e)&&m(o,e,t[e]);if(p)for(var e of p(t))g.call(t,e)&&m(o,e,t[e]);return o};var u=(o,t,e)=>new Promise((a,s)=>{var n=i=>{try{r(e.next(i))}catch(d){s(d)}},l=i=>{try{r(e.throw(i))}catch(d){s(d)}},r=i=>i.done?a(i.value):Promise.resolve(i.value).then(n,l);r((e=e.apply(o,t)).next())});var y=`${window.location.origin}/api/`;window.$.fn.dataTable&&(window.$.fn.dataTable.ext.errMode="none");var v=class extends HTMLElement{constructor(){super(),this.initState(),this.id=`table_${this.state.tableId}`,this.table=null}initState(){this.state={tableId:this.getAttribute("table_id")||"default",quarterNumber:this.getAttribute("q")||"1",data:[],columns:[],isError:!1,isLoading:!0,quarter:{current:"1",prev:null,next:null},dataTableSettings:{responsive:!0,searching:!0,ordering:!0,paging:!0,destroy:!0,debug:!1,language:{search:"Search table:"}}}}static get observedAttributes(){return["table_id","q"]}connectedCallback(){this.setupLazyLoad()}fetchData(){return u(this,null,function*(){try{let t=[{name:"Tiger Nixon",position:"System Architect",office:"Edinburgh",age:61,startDate:"2011/04/25",salary:"$320,800"},{name:"Garrett Winters",position:"Accountant",office:"Tokyo",age:63,startDate:"2011/07/25",salary:"$170,750"},{name:"Ashton Cox",position:"Junior Technical Author",office:"San Francisco",age:66,startDate:"2009/01/12",salary:"$86,000"},{name:"Cedric Kelly",position:"Senior Javascript Developer",office:"Edinburgh",age:22,startDate:"2012/03/29",salary:"$433,060"},{name:"Airi Satou",position:"Accountant",office:"Tokyo",age:33,startDate:"2008/11/28",salary:"$162,700"}],e=[{data:"name",title:"Name"},{data:"position",title:"Position"},{data:"office",title:"Office"},{data:"age",title:"Age"},{data:"startDate",title:"Start Date"},{data:"salary",title:"Salary"}];this.setState({data:t,columns:e})}catch(t){console.error("Error fetching table data:",t),this.setState({isError:!0})}finally{this.setState({isLoading:!1})}})}setState(t){this.state=h(h({},this.state),t),this.render()}renderSpinner(){return`<div class="spinner-container" role="status">
```

Biome (1.9.4)

[error] This generator function doesn't contain yield.



(lint/correctness/useYield)